test/test_gem2rpm.rb

```diff
@@ -36,6 +36,19 @@ def test_show_message_for_obj
     assert_equal("foo\n\nbar\n", @out.string)
   end
 
+  def test_show_templates
+    Gem2Rpm.show_templates(@out)
+    out_str = @out.string
```

Comment:
The merit for my implementation is that we can avoid @out.string method with repeat, to reduce the cost for the procedure.
Though the cost to call the the method may be small.
However I can modify my code for your idea.